rsocket-core/src/main/java/io/rsocket/resume/ResumedFramesCalculator.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2015-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.rsocket.resume;
+
+import reactor.core.publisher.Mono;
+
+interface ResumedFramesCalculator {
+
+  ResumedFramesCalculator ofClient = new ClientResumedFramesCalculator();
+  ResumedFramesCalculator ofServer = new ServerResumedFramesCalculator();
+
+  Mono<Long> calculate(ResumptionState local, ResumptionState remote);
+
+  class ClientResumedFramesCalculator implements ResumedFramesCalculator {
```

Comment:
Is there are reason this is async? It doesn't look like its doing anything sync

Reply:
It captures error if resumption is not possible with states provided, which then is used in RSocket sessions like https://github.com/mostroverkhov/rsocket-java/blob/42314ce746aab0c7908a677f72fabbe8c872b54d/rsocket-core/src/main/java/io/rsocket/resume/ServerRSocketSession.java#L105

Reply:
Sorry - this comment was supposed to be on `   Mono<Long> calculate(ResumptionState local, ResumptionState remote); ` 

I still don't still don't see why this async?